server/src/services/plugin-execution.service.ts

```diff
@@ -0,0 +1,214 @@
+  jwtToken: string;
+  asset: Asset;
+  triggerConfig: TriggerConfig | null;
+}
+
+@Injectable()
+export class PluginExecutionService extends BaseService {
+  private loadedPlugins: Map<string, ExtismPlugin> = new Map();
+  private hostFunctions!: PluginHostFunctions;
+
+  @OnEvent({ name: 'AppBootstrap' })
+  async onBootstrap() {
+    this.hostFunctions = new PluginHostFunctions(
+      this.assetRepository,
+      this.albumRepository,
+      this.accessRepository,
+      this.logger,
+    );
+    await this.loadCorePlugins();
+  }
+
+  private async loadCorePlugins() {
+    const plugins = await this.pluginRepository.getAllPlugins();
+    for (const plugin of plugins) {
+      try {
+        const pluginPath = resolve(process.cwd(), '..', plugin.wasmPath);
+        this.logger.debug(`Loading plugin: ${plugin.name} from ${pluginPath}`);
+
+        const extismPlugin = await newPlugin(pluginPath, {
+          useWasi: true,
+          functions: this.hostFunctions.getHostFunctions(),
+        });
+
+        this.loadedPlugins.set(plugin.id, extismPlugin);
+        this.logger.log(`Successfully loaded plugin: ${plugin.name}`);
+      } catch (error) {
+        this.logger.error(`Failed to load plugin ${plugin.name}:`, error);
+      }
+    }
+  }
+
+  @OnEvent({ name: 'AssetCreate' })
+  async handleAssetCreate({ asset }: ArgOf<'AssetCreate'>) {
+    const trigger = pluginTriggers.find((trigger) => trigger.type === PluginTriggerType.AssetCreate);
+    if (!trigger) {
+      this.logger.debug('No trigger found for asset_uploaded');
+      return;
+    }
+
+    const workflows = await this.workflowRepository.getWorkflowsByTrigger(PluginTriggerType.AssetCreate);
+    if (workflows.length === 0) {
+      this.logger.debug('No enabled workflows found for asset_uploaded trigger');
+      return;
+    }
+
+    const jobs: JobItem[] = workflows.map((workflow) => ({
+      name: JobName.AssetCreateWorkflow,
+      data: {
+        workflowId: workflow.id,
+        assetId: asset.id,
+      },
+    }));
+
+    await this.jobRepository.queueAll(jobs);
+    this.logger.debug(`Queued ${jobs.length} workflow execution jobs for asset ${asset.id}`);
+  }
+
+  @OnJob({ name: JobName.AssetCreateWorkflow, queue: QueueName.Workflow })
+  async handleAssetCreateWorkflow({ workflowId, assetId }: JobOf<JobName.AssetCreateWorkflow>): Promise<JobStatus> {
+    try {
+      const asset = await this.assetRepository.getById(assetId);
+      if (!asset) {
+        this.logger.error(`Asset ${assetId} not found`);
+        return JobStatus.Failed;
+      }
+
+      const token = JSON.stringify({ userId: asset.ownerId });
+
+      await this.executeAssetCreateWorkflow(token, workflowId, asset);
+      return JobStatus.Success;
+    } catch (error) {
+      this.logger.error(`Error executing workflow ${workflowId}:`, error);
+      return JobStatus.Failed;
+    }
+  }
+
+  private async executeAssetCreateWorkflow(jwtToken: string, workflowId: string, asset: Asset) {
```

Comment:
this method doesn't actually have anything specific in it about besides the fact that context.asset is set. I think we should construct this object in the `onWorkflowRun` job and this code is run for all workflow trigger types. The parent job just needs to construct the `context` that will be passed in here.